components/org.wso2.bfsi.consent.management.extensions/src/main/java/org/wso2/bfsi/consent/management/extensions/validate/impl/DefaultConsentValidator.java

```diff
@@ -0,0 +1,367 @@
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpStatus;
+import org.wso2.bfsi.consent.management.dao.models.AuthorizationResource;
+import org.wso2.bfsi.consent.management.dao.models.DetailedConsentResource;
+import org.wso2.bfsi.consent.management.extensions.common.ConsentException;
+import org.wso2.bfsi.consent.management.extensions.common.ConsentExtensionConstants;
+import org.wso2.bfsi.consent.management.extensions.common.ResponseStatus;
+import org.wso2.bfsi.consent.management.extensions.validate.ConsentValidator;
+import org.wso2.bfsi.consent.management.extensions.validate.model.ConsentValidateData;
+import org.wso2.bfsi.consent.management.extensions.validate.model.ConsentValidationResult;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Consent validator default implementation.
+ */
+public class DefaultConsentValidator implements ConsentValidator {
+
+    private static final Log log = LogFactory.getLog(DefaultConsentValidator.class);
+    private static final String ACCOUNTS_REGEX = "/accounts/[^/?]*";
+    private static final String TRANSACTIONS_REGEX = "/accounts/[^/?]*/transactions";
+    private static final String BALANCES_REGEX = "/accounts/[^/?]*/balances";
+    private static final String COF_SUBMISSION_PATH = "/funds-confirmations";
+    private static final String PERMISSION_MISMATCH_ERROR = "Permission mismatch. Consent does not contain necessary " +
+            "permissions";
+    private static final String INVALID_URI_ERROR = "Path requested is invalid";
+    private static final String CONSENT_EXPIRED_ERROR = "Provided consent is expired";
+    @Override
+    public void validate(ConsentValidateData consentValidateData, ConsentValidationResult consentValidationResult)
+            throws ConsentException {
+
+        String uri = consentValidateData.getRequestPath();
+        JSONObject receiptJSON;
+        try {
+            receiptJSON = (JSONObject) (new JSONParser(JSONParser.MODE_PERMISSIVE)).
+                    parse(consentValidateData.getComprehensiveConsent().getReceipt());
+
+        } catch (ParseException e) {
+            log.error(e.getMessage().replaceAll("[\n\r]", ""));
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, "Exception occurred while validating" +
+                    " permissions");
+        }
+
+        //User Validation
+        String userIdFromToken = consentValidateData.getUserId();
+        boolean userIdMatching = false;
+        ArrayList<AuthorizationResource> authResources = consentValidateData.getComprehensiveConsent()
+                .getAuthorizationResources();
+        for (AuthorizationResource resource : authResources) {
+            if (userIdFromToken.contains(resource.getUserID())) {
+                userIdMatching = true;
+                break;
+            }
+        }
+
+        //TODO:
+//        if (!userIdMatching) {
```

Comment:
Here we can consider uncomment this logic to validate userID matching right?

Reply:
We have commented this due to an issue in username stored in our database. It was stored as uuid. once that issue is sorted, I will uncomment this.